inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/heartbeat/HeartbeatManager.java

```diff
@@ -274,11 +281,13 @@ private void evictClusterNode(HeartbeatMsg heartbeat) {
             return;
         }
 
-        for (int i = 0; i < ports.length; i++) {
+        for (int i = 0; i < ips.length; i++) {
             // deep clone the heartbeat
             HeartbeatMsg heartbeatMsg = JsonUtils.parseObject(JsonUtils.toJsonByte(heartbeat), HeartbeatMsg.class);
             assert heartbeatMsg != null;
-            heartbeatMsg.setPort(ports[i].trim());
+            if (StringUtils.isNotBlank(heartbeat.getPort())) {
+                heartbeatMsg.setPort(ports[i].trim());
```

Comment:
`ports` may be null even if `heartbeat.getPort()` is not blank.

Reply:
ports is not null when heartbeat.getPort() is not blank.

Reply:
`ports` is set to null when `ports.length < ips.length`

Reply:
Fixed.